osbuild-composer.spec

```diff
@@ -307,6 +307,7 @@ cd $PWD/_build/src/%{goipath}
 %doc README.md
 %{_mandir}/man7/%{name}.7*
 %{_unitdir}/osbuild-composer.service
+%{_unitdir}/osbuild-dnf-json.socket
```

Comment:
This is a packaging bug. The `.socket` unit should be part of the `dnf-json` sub-package, as the `.service` file is.

Reply:
I don't understand how `.spec` is structured yet. I reproduced what was already done in the file for the sub-package `worker` and applied it to `dnf-json`. Why is the `.socket` unit not with the sub-package `worker`, is this a mistake too ? Should I correct the two ?

Reply:
No, those are fine. This is a bit misleading, but let me explain it.

In general the `.socket` unit starts a specific `.service` unit. The worker socket units are actually starting `osbuild-composer.service`, because it is the server's socket and the worker actually connects to it. So in this case this is correct. Worker does not listen on any socket, it connects to composer. In `dnf-json` case, the socket starts the service with the same base name and therefore it should be in the same binary RPM. In general everything would work just fine even now, because due to RPM dependencies the `dnf-json` sub-package will be pulled in when the main composer RPM is installed. But if you would install just the `dnf-json` sub-package, you would not be able to connect to it because systemd would not listen on the appropriate socket and would not start the service.

Reply:
Thanks a lot for the explanation!